dashboard/src/app/[tenantId]/components/OverflowText.tsx

```diff
@@ -0,0 +1,60 @@
+'use client'
+import { FC, useEffect, useRef, useState } from 'react'
+import { cn } from '@/components/utils'
+import { Button } from '@/components/ui/button'
+import { ChevronRight } from 'lucide-react'
+import { Dialog, DialogContent, DialogTrigger } from '@/components/ui/dialog'
+import { CopyButton } from './CopyButton'
+import { tryFormatAsJson } from '@/app/utils/tryFormatAsJson'
+
+type OverflowTextProps = {
+  text: string
+  className?: string
+}
+
+export const OverflowText: FC<OverflowTextProps> = ({ text, className }) => {
+  const textRef = useRef<HTMLDivElement>(null)
+  const [isOverflowing, setIsOverflowing] = useState(false)
+
+  useEffect(() => {
+    const element = textRef.current
+    if (element) {
+      setIsOverflowing(element.scrollWidth > element.clientWidth)
+    }
+  }, [text])
+
+  const formattedText = tryFormatAsJson(text)
```

Comment:
Why not use the same function that the variables use in the `taskRun` modal?

Reply:
can u link me the location of the function plz